### ISSUE TITLE
menpo.io.export_landmark_file fails on ljson export

### ISSUE THREAD

**Issue by mkutny** [menpo/menpo#741]
Apparently `_export(obj, fp)` always opens a file as 'wb' while `json.dump(obj, fp)` which is subsequently called "always produces str objects, not bytes objects. Therefore, fp.write() must support str input".

Below is a traceback:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-804876baea20> in <module>()
----> 1 menpo.io.export_landmark_file(alex.landmarks['bb'], '/Users/max/landmarks/alex.ljson', 'ljson', True)

/Users/max/miniconda3/envs/menpo/lib/python3.5/site-packages/menpo/io/output/base.py in export_landmark_file(landmark_group, fp, extension, overwrite)
     50         (the output type is not supported).
     51     """
---> 52     _export(landmark_group, fp, landmark_types, extension, overwrite)
     53 
     54 

/Users/max/miniconda3/envs/menpo/lib/python3.5/site-packages/menpo/io/output/base.py in _export(obj, fp, extensions_map, extension, overwrite, protocol)
    363     else:
    364         # You MUST provide an extension if a file handle is given
--> 365         if extension is None:
    366             raise ValueError('An export file extension must be provided if a '
    367                              'file-like object is passed.')

/Users/max/miniconda3/envs/menpo/lib/python3.5/site-packages/menpo/io/output/landmark.py in ljson_exporter(landmark_group, file_handle, **kwargs)
     46 
     47     return json.dump(lg_json, file_handle, indent=4, separators=(',', ': '),
---> 48                      sort_keys=True, allow_nan=False)
     49 
     50 

/Users/max/miniconda3/envs/menpo/lib/python3.5/json/__init__.py in dump(obj, fp, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    177     # a debuggability cost
    178     for chunk in iterable:
--> 179         fp.write(chunk)
    180 
    181 

TypeError: a bytes-like object is required, not 'str'
```

**Comment by patricksnape** [comment]
I actually fixed that [here](https://github.com/menpo/menpo/blob/master/menpo/io/output/landmark.py#L53). But thanks for reporting the issue! We will make a new release ASAP so that you don't have to keep suffering this. In the meantime, you could always install the master build of menpo by using the command:

`conda install -c menpo/channel/master -c menpo menpo`

**Comment by mkutny** [comment]
Thanks for the quick response!

As a temporary workaround I used
```
with open('alex.ljson', 'wt') as f:
    menpo.io.export_landmark_file(alex.landmarks['bb'], f, 'ljson', True)
```

instead of
`menpo.io.export_landmark_file(alex.landmarks['bb'], 'alex.ljson', 'ljson', True)`

and it worked fine.

**Comment by patricksnape** [comment]
Yes that will work too! Good workaround :smile:

**Comment by patricksnape** [comment]
Should be fixed in 0.7.5
